pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/minion/PinotTaskManager.java

```diff
@@ -475,6 +476,14 @@ public void registerTaskGenerator(PinotTaskGenerator taskGenerator) {
     _taskGeneratorRegistry.registerTaskGenerator(taskGenerator);
   }
 
+  /**
+   * Registers a task segment group manager.
+   * <p>This method can be used to plug in custom task segment group manager.
+   */
+  public void registerTaskSegmentGroupManager(PinotTaskSegmentGroupManager taskSegmentGroupManager) {
```

Comment:
Instead of exposing this function for registering and setting `taskSegmentGroupManager` from the top level task manager and feed this all the way to the task generator, I can think of another approach which would not require interface changes:

1. Pass the factory class name for `TaskSegmentGroupManager` as part of the task config for Merge&Rollup task. 
2. initialize the `TaskSegmentGroupManager` object based on the factory class name


This approach is what we use for Kafka client factory in the stream config.

Reply:
Good point, updated. PTAL.